modules/lti/src/main/resources/tools/series/series.js

```diff
@@ -60,49 +61,67 @@ function i18n(key) {
   return lang[key];
 }
 
-function getSeries() {
+function getParam(name) {
   const urlParams = new URLSearchParams(window.location.search);
-  if (urlParams.has('series')) {
-    return urlParams.get('series');
+  if (urlParams.has(name)) {
+    return urlParams.get(name);
   }
   return '';
 }
 
 function loadPage(page) {
-
   var limit = 15,
       offset = (page - 1) * limit,
-      series = getSeries(),
+      series = getParam('series'),
+      seriesName = getParam('series_name'),
+      isInstructor = false,
+      roles = [],
+      episodes,
       url = '/search/episode.json?limit=' + limit + '&offset=' + offset;
 
   currentpage = page;
 
   // attach series query if a series is requested
   if (series) {
     url += '&sid=' + series;
+  } else if (seriesName) {
+    url += '&sname=' + seriesName;
   }
 
   // load spinner
   $('main').html($('#template-loading').html());
 
-  $.getJSON(url, function( data ) {
-    data = data['search-results'];
+  $.when(
+    $.getJSON('/lti', function( data ) {
+      if (data.roles === undefined) {
+        isInstructor = false;
+      } else {
+        roles = data['roles'].split(',');
+        isInstructor = roles.includes('Instructor');
+      }
+    }),
+    $.getJSON(url, function( data ) {
+      episodes = data['search-results'];
+    })
+  ).then(function() {
     var rendered = '',
         results = [],
-        total = parseInt(data.total);
+        total = parseInt(episodes.total);
 
     if (total > 0) {
-      results = Array.isArray(data.result) ? data.result : [data.result];
+      results = Array.isArray(episodes.result) ? episodes.result : [episodes.result];
     }
 
     for (var i = 0; i < results.length; i++) {
       var episode = results[i],
           i18ncreator = Mustache.render(i18n('CREATOR'), {creator: episode.dcCreator}),
-          template = $('#template-episode').html(),
+          template = isInstructor ? $('#template-episode-admin').html() : $('#template-episode-user').html(),
           tpldata = {
             player: player + episode.id,
+            uid: episode.id,
             title: episode.dcTitle,
             i18ncreator: i18ncreator,
+            hasDeletion: getParam('deletion') == 'true',
```

Comment:
This parameter isn't documented

Reply:
Not sure _where_ you want it to be documented. It's described in `ltimodule.md`. Should I add a source code comment?